taxonomy-editor-frontend/src/components/useFetch.jsx

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from "react";
+
+const useFetch = (url) => {    
+    const [data, setData] = useState(null)
+    const [isPending, setIsPending] = useState(true)
+    const [error, setError] = useState(null);

```

Comment:
these two (isPending, error) could be a variable called status, so the 'status' can be a string: 'error', 'pending', etc. Even better to have a single state that contains both the data and the status

Reply:
@nobeeakon I did most of my development using these variables separately. If I change it here, I would need refactor in a lot of places. I think it might be better to keep them separately instead, cause I would anyways make it into separate variables wherever I use them. Do you agree?

Reply:
mmm well, yeah I see the point, I think that changes are somehow inevitable when reviewing. Although I think it may work this way, and not modifying it would make us move faster (less things to revisit in the next commits),  think that having everything in a single state and having a `status` instead of `booleans` makes it easier to understand the logic, for example, in this case, you may need to always set three independent states. Lets say you first have an error and then the url changes and you have a success, currently you would have an error message from a previous request and a piece of data. So, if, for example in the component you first check if there's an error and display it, you could end up always showing it. As said, I think if we keep it like this it could work, but think it is good to keep things as clean/clear as we can. Also, I guess `error` could be renamed to `errorMessage`

Reply:
Yeah, I can rename that. Also, I'm not sure that would happen since the state variables displayed are totally dependent on the hook and if the hook runs, the states get reset. Is my intuition right?

Reply:
sorry @aadarsh-ram , not sure I understood :s

Reply:
> Lets say you first have an error and then the url changes and you have a success, currently you would have an error message from a previous request and a piece of data. So, if, for example in the component you first check if there's an error and display it, you could end up always showing it.

@nobeeakon In this case, all the states that are stored in the useFetch hook, are reset each time whenever the hook is called. So, if a URL changes, the useFetch hook will be called and previous states will be reset. So, I'll not be showing any previous errors if any. So, having separate variables won't result in any disadvantage I think. Does this make sense?